### PR TITLE
kleopatra: fix broken build

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

Kleopatra broke in #187453, this patch is also applied by Arch. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/kde/kleopatra.nix [hunk 1]
```diff
@@ -18,4 +18,11 @@ mkDerivation {
     boost gpgme kcmutils kdbusaddons kiconthemes kitemmodels kmime
     knotifications kwindowsystem kxmlgui libkleo kcrash
   ];
+
+  patches = [
```
Comment (SuperSandro2000) [inline pkgs/applications/kde/kleopatra.nix:22]: the attribute ordering in this file is weird. It should look like:

```nix
mkDerivation {
  pname = "kleopatra";

  patches = [
    (fetchpatch {
      url = "https://invent.kde.org/pim/kleopatra/-/commit/87d8b00d4b2286489d5fadc9cfa07f1d721cdfe3.patch";
      sha256 = "sha256-s1tXB7h0KtFwwZHx8rhpI0nLZmwhWAiraHEF3KzncMc=";
    })
  ];

  nativeBuildInputs = [ extra-cmake-modules kdoctools ];

  buildInputs = [
    boost gpgme kcmutils kdbusaddons kiconthemes kitemmodels kmime
    knotifications kwindowsystem kxmlgui libkleo kcrash
  ];

  meta = {
    homepage = "https://apps.kde.org/kleopatra/";
    description = "Certificate manager and unified crypto GUI";
    license = with lib.licenses; [ gpl2 lgpl21 fdl12 ];
    maintainers = kdepimTeam;
  };
}
```

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 188974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.kleopatra</li>
    <li>libsForQt512.kmail</li>
    <li>libsForQt514.kleopatra</li>
    <li>libsForQt514.kmail</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>libsForQt5.kleopatra (libsForQt515.kleopatra ,plasma5Packages.kleopatra)</li>
    <li>libsForQt5.kmail (libsForQt515.kmail ,plasma5Packages.kmail)</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 188974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.kleopatra</li>
    <li>libsForQt512.kmail</li>
    <li>libsForQt514.kleopatra</li>
    <li>libsForQt514.kmail</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>libsForQt5.kleopatra (libsForQt515.kleopatra ,plasma5Packages.kleopatra)</li>
    <li>libsForQt5.kmail (libsForQt515.kmail ,plasma5Packages.kmail)</li>
  </ul>
</details>

File: pkgs/applications/kde/kleopatra.nix [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
   extra-cmake-modules, kdoctools,
   boost, gpgme, kcmutils, kdbusaddons, kiconthemes, kitemmodels, kmime,
   knotifications, kwindowsystem, kxmlgui, libkleo, kcrash
 }:
 
 mkDerivation {
   pname = "kleopatra";
+
+  patches = [
+    (fetchpatch {
```
Comment (peterhoeg) [inline pkgs/applications/kde/kleopatra.nix:12]: It's not in any way critical, but typically we either name the patch using the name attribute like this `name = "fix_build.patch";` or put a comment to show why we are patching.

Comment (ulrikstrid) [reply pkgs/applications/kde/kleopatra.nix:12]: I totally forgot. The reason is that the released version is not compatible with the version of pgpme we (and arch) have.

**Pull request merged**